File: src/main/java/br/com/fabricadeprogramador/ws/controller/VeiculoController.java
package br.com.fabricadeprogramador.ws.controller;

import java.util.Collection;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import br.com.fabricadeprogramador.ws.model.Veiculo;
import br.com.fabricadeprogramador.ws.service.VeiculoService;

@RestController
public class VeiculoController {
	
	@Autowired
	private VeiculoService veiculoService;
	
	@RequestMapping(method=RequestMethod.POST, value="/veiculos", consumes=MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity<Veiculo> cadastrarVeiculo(@RequestBody Veiculo veiculo) {
		Veiculo veiculoCadastrado = veiculoService.cadastrar(veiculo);
		return new ResponseEntity<Veiculo>(veiculoCadastrado, HttpStatus.CREATED);
	}
	
	@RequestMapping(method=RequestMethod.GET, value="/veiculos", produces=MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity<Collection<Veiculo>> buscarTodosVeiculos() {
		Collection<Veiculo> veiculosRetornados = veiculoService.buscarTodos();
		return new ResponseEntity<>(veiculosRetornados, HttpStatus.OK);
	}
}

File: bin/src/main/resources/static/js/controller/veiculo-controller.js
appCliente.controller("veiculoController", function($scope, $http) {
	$scope.veiculos = [];
	$scope.veiculo = {};
	
	$scope.carregarVeiculos = function() {
		$http({method:'GET', url:'/veiculos'}).then(
			function(response){
				$scope.veiculos = response.data;
			}, 
			function(response){
				console.log(response.data);
				console.log(response.status);
			}
		);
	}
	
	$scope.salvarVeiculo = function () {
		$http({method:'POST', url:'/veiculos', data: $scope.veiculo}).then(
			function(response){
				$scope.carregarVeiculos();
				$scope.cancelar();
			}, 
			function(response){
				console.log(response.data);
				console.log(response.status);
			}
		);
	}
	
	$scope.cancelar = function () {
		$scope.veiculo = {};
	}
	
	$scope.carregarVeiculos();
});